site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,263 @@
+
+```js
+if (HTMLScriptElement.supports && HTMLScriptElement.supports('speculationrules')) {
+  console.log('Your browser supports the Speculation rules API.');
+}
+```
+
+Speculation rules can be:
+
+* Statically inserted in the page's HTML. For example a news media site, or a blog may prerender the newest article, if that is often the next navigation for a large proportion of users.
+* Dynamically inserted into the page by JavaScript. This could be based on application logic, personalized to the user, or on certain user actions such as hovering over, or clicking down on a link—as many libraries have done in the past with `preconnect`, `prefetch`, or even `preload`.
+
+Speculation rules can be added anywhere in the main page and do not need to be in the `<head>` element. Speculation rules in subframes are not acted upon, and speculation rules in prerendered pages, are only acted upon once the page is activated.
+
+Below is a simple example of adding a `prerender` speculation rule:
+
+```js
+if (HTMLScriptElement.supports &&
+    HTMLScriptElement.supports('speculationrules')) {
+    const specScript = document.createElement("script");
+    specScript.type = 'speculationrules';
+    specRules = {
+      'prerender': [
+        {
+          source: 'list',
+          urls: ['/next.html'],
+        },
+      ],
+    };
+    specScript.textContent = JSON.stringify(specRules);
+    console.log('added speculation rules to: next.html');
+    document.body.append(specScript);
+}
+```
+
+You can view a demo of Speculation Rules prerendering on [this demo page](https://prerender-demos.glitch.me/).
+
+## Detecting and disabling prerendering
+
+Prerender is usually a positive experience for users as it allows fast page rendering—often instant. This benefits both the user, and the site owner, since prerendered pages allow a better user experience that may be difficult to achieve otherwise.
+
+However, there may be instances when you do not wish prerendering of pages to happen, for example when pages change state—either based on the initial request, or based on JavaScript executing on the page.
+
+### Enabling or disabling prerender in Chrome
+
+Prerender is only enabled for those Chrome users with the “Preload pages for faster browsing and searching” setting in `chrome://settings/cookies/`. Additionally, prerender is also disabled on low-memory devices, or if the operating system is in data-saver or battery-saver modes.
+
+### Detecting prerender server-side
+
+Prerendered pages will be sent with the following HTTP header:
+
+```http
+Sec-Purpose: prefetch;prerender
+```
+
+Servers can respond based on this header, to log prerender requests, return different content, or prevent the prerender from happening. If a non-success response code is returned (that is, not a 200 or a 304), then the page will not be prerendered. So this is one way to prevent pages you absolutely do not want prerendered, from prerendering.
```

Comment:
Would it be worth showing some back-end application logic as an example to prevent a prerender? Like an Apache rule or even a PHP/Express example that throws the proper response code if the developer doesn't want a page to be prerendered?

Reply:
I'm going to leave that out as we're trying to discourage that as much as possible.